src/Stryker.Core/Stryker.Core/Mutants/MutantOrchestrator.cs

```diff
@@ -62,12 +64,150 @@ public IEnumerable<Mutant> GetLatestMutantBatch()
+        {
+            if (expressionStatementNeedingIf.Contains(expressionStatement.Expression.GetType()))
+            {
+                var expressionCopy = expressionStatement.TrackNodes(expressionStatement, expressionStatement.Expression);
+                SyntaxNode result = MutateSubExpressionWithIfStatements(expressionStatement, expressionCopy, expressionStatement.Expression);
+                if (!(expressionStatement.Expression is AssignmentExpressionSyntax))
+                {
+                    return result;
+                }
+
+                return result.ReplaceNode(result.GetCurrentNode(expressionStatement), MutateExpressions(expressionStatement));
+            }
+
+            return MutateExpressions(expressionStatement);
+        }
+
+
+        private SyntaxNode MutateIfStatement(IfStatementSyntax ifStatement)
+        {
+            var mutatedIf = ifStatement.Else != null
+                ? ifStatement.TrackNodes(ifStatement.Condition, ifStatement.Statement, ifStatement.Else)
+                : ifStatement.TrackNodes(ifStatement.Condition, ifStatement.Statement);
+
+            if (!ifStatement.Condition.DescendantNodes().Any(x =>
+                x.IsKind(SyntaxKind.DeclarationExpression) || x.IsKind(SyntaxKind.DeclarationPattern)))
+            {
+                mutatedIf = mutatedIf.ReplaceNode(mutatedIf.GetCurrentNode(ifStatement.Condition),
+                    MutateWithConditionalExpressions(ifStatement.Condition));
+            }
+
+            if (ifStatement.Else != null)
+            {
+                mutatedIf = mutatedIf.ReplaceNode(mutatedIf.GetCurrentNode(ifStatement.Else), Mutate(ifStatement.Else));
+            }
+
+            return mutatedIf.ReplaceNode(mutatedIf.GetCurrentNode(ifStatement.Statement), Mutate(ifStatement.Statement));
+        }
+
+        private SyntaxNode MutateForStatement(ForStatementSyntax forStatement)
+        {
+            // for needs special treatments for its incrementors
+            StatementSyntax trackedCopy = forStatement;
+
+            foreach (var incrementor in forStatement.Incrementors)
+            {
+                trackedCopy = MutateSubExpressionWithIfStatements(forStatement, trackedCopy, incrementor);
+            }
+
+            var originalFor = trackedCopy;
+            if (trackedCopy != forStatement)
+            {
+                var scan = trackedCopy;
+                while (scan is IfStatementSyntax mutantIf)
+                {
+                    scan = ((BlockSyntax) mutantIf.Else.Statement).Statements[0];
+                }
+
+                originalFor = (ForStatementSyntax) scan;
+            }
+
+            // now we generate a mutant for the remainder of the for statement
+            var mutatedFor = forStatement.TrackNodes(forStatement.Condition, forStatement.Statement);
+            mutatedFor = mutatedFor.ReplaceNode(mutatedFor.GetCurrentNode(forStatement.Condition),
```

Comment:
`mutatedFor.GetCurrentNode` in all other methods we call the tracked nodes `Copy`. Why not here?

Reply:
Well, I think it is more explicit as it expresses intent, vs copy which is broader. But I did not rename every other usage. So I think every other usage should be properly renamed, but I try to avoid endless naming discussions, so you tell me what you want